purchases_ui_flutter/android/src/main/kotlin/com/revenuecat/purchases_ui_flutter/views/PaywallView.kt

```diff
@@ -23,8 +29,41 @@ internal class PaywallView(
     override fun dispose() {}
 
     init {
+        methodChannel = MethodChannel(messenger, "com.revenuecat.purchasesui/PaywallView/$id")
+        methodChannel.setMethodCallHandler(this)
         val offeringIdentifier = creationParams["offeringIdentifier"] as String?
         nativePaywallView = NativePaywallView(context,)
+        nativePaywallView.setPaywallListener(object : PaywallListenerWrapper() {
+            override fun onPurchaseStarted(rcPackage: Map<String, Any?>) {
+                methodChannel.invokeMethod("onPurchaseStarted", rcPackage)
+            }
+
+            override fun onPurchaseCompleted(customerInfo: Map<String, Any?>, storeTransaction: Map<String, Any?>) {
+                methodChannel.invokeMethod(
+                    "onPurchaseCompleted",
+                    mapOf("customerInfo" to customerInfo, "storeTransaction" to storeTransaction)
+                )
+            }
+
+            override fun onPurchaseError(error: Map<String, Any?>) {
+                methodChannel.invokeMethod("onPurchaseError", error)
+            }
+
+            override fun onRestoreCompleted(customerInfo: Map<String, Any?>) {
+                methodChannel.invokeMethod("onRestoreCompleted", customerInfo)
+            }
+
+            override fun onRestoreError(error: Map<String, Any?>) {
+                methodChannel.invokeMethod("onRestoreError", error)
+            }
+        })
+        // TODO add to constructor
         nativePaywallView.setOfferingId(offeringIdentifier)
     }
+
+    override fun onMethodCall(methodCall: MethodCall, result: MethodChannel.Result) {
```

Comment:
We could maybe add a comment saying that we currently don't have any communication in this channel from dart to native, so this can be empty.